### PR TITLE
Add an agent selector to IT Hygiene application

### PR DESCRIPTION
**Pull request by Desvelao**
### Description
This pull request adds an agent selector to the IT Hygiene application.

Changes:
- Wrapped the IT Hygiene application with a required selection of agent
- Added a Explore agent button to the IT Hygiene application
- The Explore agent button of alert modules now display the global agent
  selector instead of the local one.
- Removed the local agent selector modal of Explore agent button on alert
  modules
- Removed the related action creator and reducer case for SHOW_EXPLORE_AGENT_MODAL
- Refactor MITRE ATT&CK panel to allow the reactivity when changing the selected agent
  - Rename MITRE to MITRE ATT&CK in the panel
- Rename mitre_top to mitre-top according to the file naming convention
- Add reactivity to the FIM: Recent events table when changing the
  selected agent

### Issues Resolved
[List any issues this PR will resolve]

### Evidence
image
image
image
image

### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome |
| --- |  --- |
| Without selected agent, access to the IT Hygiene application. A prompt should indicate the requirement of selecting an agent. | :black_circle: |
| Selecting an agent in the IT Hygiene agent selection prompt should display the data related to the agent | :black_circle: |
| Use the Explore agent button of IT Hygiene to unpin the agent. The prompt should display it is needed a selected agent. | :black_circle: |
| Use the Explore agent button of IT Hygiene to change the selected agent. The view and displayed data should be related to the selected agent. Ensure the data panel are updated with the new information. | :black_circle: |
| Ensure the Explore agent button on the legacy modules (Dashboard, Events, Panel, Inventory tabs) allows to select an agent and the search URL paramenter is changed accordingly (agent, agentId). | :black_circle: |
| Ensure the Explore agent button on the legacy modules allows to unpint an agent and the search URL paramenter is removed accordingly (agent, agentId). | :black_circle: |
| Pin an agent and navigate to Inventory data, Stats and Configuration. Ensure the data is related to the pinned agent. | :black_circle: |

**Details**
<details>
<summary>:black_circle: Without selected agent, access to the IT Hygiene application. A prompt should indicate the requirement of selecting an agent.</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Selecting an agent in the IT Hygiene agent selection prompt should display the data related to the agent</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Use the Explore agent button of IT Hygiene to unpin the agent. The prompt should display it is needed a selected agent.</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Use the Explore agent button of IT Hygiene to change the selected agent. The view and displayed data should be related to the selected agent. Ensure the data panel are updated with the new information.</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Ensure the Explore agent button on the legacy modules (Dashboard, Events, Panel, Inventory tabs) allows to select an agent and the search URL paramenter is changed accordingly (agent, agentId).</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Ensure the Explore agent button on the legacy modules allows to unpint an agent and the search URL paramenter is removed accordingly (agent, agentId).</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Pin an agent and navigate to Inventory data, Stats and Configuration. Ensure the data is related to the pinned agent.</summary>

Chrome - :black_circle:

</details>

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: plugins/main/public/controllers/agent/agents.js [hunk 1]
```diff
@@ -63,6 +63,9 @@ export class AgentsController {
     this.csvReq = csvReq;
     this.wazuhConfig = new WazuhConfig();
     this.genericReq = GenericRequest;
+    this.$scope.$watch('tab', function (newValue, oldValue) {
+      console.log('tab changed', { newValue, oldValue });
+    });
```
Comment (Desvelao) [inline plugins/main/public/controllers/agent/agents.js:68]: **nitpick**: remove this code.

**Comment by yenienserrano** [comment]
### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome |
| --- |  --- |
| Without selected agent, access to the IT Hygiene application. A prompt should indicate the requirement of selecting an agent. | :black_circle: |
| Selecting an agent in the IT Hygiene agent selection prompt should display the data related to the agent | :black_circle: |
| Use the Explore agent button of IT Hygiene to unpin the agent. The prompt should display it is needed a selected agent. | :black_circle: |
| Use the Explore agent button of IT Hygiene to change the selected agent. The view and displayed data should be related to the selected agent. Ensure the data panel are updated with the new information. | :black_circle: |
| Ensure the Explore agent button on the legacy modules (Dashboard, Events, Panel, Inventory tabs) allows to select an agent and the search URL paramenter is changed accordingly (agent, agentId). | :black_circle: |
| Ensure the Explore agent button on the legacy modules allows to unpint an agent and the search URL paramenter is removed accordingly (agent, agentId). | :black_circle: |

**Details**
<details>
<summary>:black_circle: Without selected agent, access to the IT Hygiene application. A prompt should indicate the requirement of selecting an agent.</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Selecting an agent in the IT Hygiene agent selection prompt should display the data related to the agent</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Use the Explore agent button of IT Hygiene to unpin the agent. The prompt should display it is needed a selected agent.</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Use the Explore agent button of IT Hygiene to change the selected agent. The view and displayed data should be related to the selected agent. Ensure the data panel are updated with the new information.</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Ensure the Explore agent button on the legacy modules (Dashboard, Events, Panel, Inventory tabs) allows to select an agent and the search URL paramenter is changed accordingly (agent, agentId).</summary>

Chrome - :black_circle:

</details>

<details>
<summary>:black_circle: Ensure the Explore agent button on the legacy modules allows to unpint an agent and the search URL paramenter is removed accordingly (agent, agentId).</summary>

Chrome - :black_circle:

File: plugins/main/public/controllers/agent/agents.js (hunk 1 above)
Comment (Desvelao) [reply plugins/main/public/controllers/agent/agents.js]: Fixed on https://github.com/wazuh/wazuh-dashboard-plugins/pull/6005/commits/cd628a40b7cc496ecc9ecd56e76313d5a9b2ce3f.

**Comment by Desvelao** [comment]
A new manual test was added to the suite.

**Review by yenienserrano — APPROVED** [review]
LGTM

**Pull request merged**